File: regexlib/python_re_test_file/regexlib_1650.py
# 1650
# ^(?:(?:((?![0-9_])[a-zA-Z0-9_]+)\.?)+)(?<!\.)$
# EXPONENT
# nums:4
# EXPONENT AttackString:""+"a"*32+"!1 __NQ"

import re
from time import perf_counter

regex = """^(?:(?:((?![0-9_])[a-zA-Z0-9_]+)\.?)+)(?<!\.)$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "a" * i * 1 + "!1 __NQ"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/2021-5-15/javascript_test_file/regexlib_1650.js
/* 1650
 * ^(?:(?:((?![0-9_])[a-zA-Z0-9_]+)\.?)+)(?<!\.)$
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"A"*32+"!1 __NQ"
 */
var REGEX = /^(?:(?:((?![0-9_])[a-zA-Z0-9_]+)\.?)+)(?<!\.)$/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + 'A'.repeat(i*1) + '!1 __NQ'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}